junit-runtime/src/main/scala/org/junit/internal/ArrayComparisonFailure.scala

```diff
@@ -5,24 +5,26 @@ package org.junit.internal
 
 object ArrayComparisonFailure
 
-class ArrayComparisonFailure(fMessage: String) extends AssertionError {
-  private var fIndices: List[Int] = Nil
+class ArrayComparisonFailure(message: String, cause: AssertionError, index: Int)
+    extends AssertionError(message, cause) {
```

Comment:
Seems like I'm going to learn something about Scala today: How is using something from the constructor parameter list different than initializing it in the constructor?

Reply:
Seems like you will: assignments to *constructor arguments* are initialized *before* the call to the super constructor. This means that when the super constructor executes (which is when `toString` apparently gets called, at some point), constructor arguments are already initiliazed, but not `val`s in the body of the class.

Reply:
Right... I didn't think about subclassing... Fair enough, that makes sense.